capstone/src/main/java/com/google/sps/servlets/calendar/EventsServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
+
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet for getting events from a club's calendar */
+@WebServlet("/events")
+public class EventsServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String clubName = request.getParameter(Constants.PROPERTY_NAME);
+    String calendarId = ServletUtil.getPropertyFromClub(clubName, Constants.CALENDAR_PROP);
+    List<Event> allEvents = getEvents(calendarId);
+
+    Gson gson = new Gson();
+    String json = gson.toJson(allEvents);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public List<Event> getEvents(String calendarId) {
+    List<Event> allEvents = new ArrayList<Event>();
+    try {
+      Calendar service = ClubServlet.getCalendarService();
+      String pageToken = null;
+      do {
+        Events events = service.events().list(calendarId).setPageToken(pageToken).execute();
```

Comment:
Is there a slightly better way to do this? For instance, not setting pageToken if it's null? And instead of always calling events.getNextPageToken() and then checking null, is there a way to check if it exists before getting it? (eg: some method like events.hasNextPageToken) Not sure super familiar with the external API, but it would be good to check this first.

Reply:
I'm not sure...it looks like the Events class doesn't have anything like events.hasNextPageToken, and they just rely on getNextPageToken to return null when it's done. This do-while method is also the general method the tutorial used to return all events too. Are there any other ways that might be cleaner?